corehq/apps/geospatial/forms.py

```diff
@@ -195,9 +208,14 @@ def clean(self):
             if not cleaned_data['target_group_count']:
                 raise ValidationError(_("Value for target group count required"))
 
-        algorithm = cleaned_data['selected_disbursement_algorithm']
-        token = cleaned_data['mapbox_token']
+        algorithm = cleaned_data.get('selected_disbursement_algorithm')
+        token = cleaned_data.get('plaintext_api_token')
         if algorithm == GeoConfig.ROAD_NETWORK_ALGORITHM and not token:
             raise ValidationError(_("Mapbox API token required"))
 
         return cleaned_data
+
+    def save(self, commit=True):
+        if self.cleaned_data.get('plaintext_api_token'):
```

Comment:
I see you changed the name of the property on the form to "plaintext_api_token". Is this if clause still needed?

Reply:
It's not technically needed, so removing.